### DISCUSSION TITLE
[FEATURE REQUEST] Preload support

### DISCUSSION THREAD

**Discussion by ladown** [webdiscus/pug-plugin#67] [Ideas]
Hello!

If there are any possibilities how to configure preload for styles or scripts. I found out a [plugin](https://github.com/vuejs/preload-webpack-plugin), but it requires HtmlWebpackPlugin to be initialized. How can i do it?

**Comment by webdiscus** [comment]
The pug-plugin is designed to replace the [html-webpack-plugin](https://github.com/jantimon/html-webpack-plugin), [pug-loader](https://github.com/webdiscus/pug-loader) and [mini-css-extract-plugin](https://github.com/webpack-contrib/mini-css-extract-plugin) completely. 
The pug-plugin will never works together with HtmlWebpackPlugin. But very useful features of HtmlWebpackPlugin can be implemented in pug-plugin.

What do the [preload-webpack-plugin](https://github.com/vuejs/preload-webpack-plugin)?
Your imported JS file will be injected into HTML  as the `link` tag like:
```html
<link rel="preload" as="script" href="chunk.31132ae6680e598f8879.js">
```

You can self this write in Pug:
```pug
link(rel="preload" as="script" href=require('./path/to/source/script.js') )
```

The generated HTML:
```html
<link rel="preload" as="script" href="script.31132ae6680e598f8879.js">
```

P.S.: Can you please describe your feature request in details what do you concretely want.

**Comment by ladown** [comment]
No doubt, that i can set preloaded resources by myself, but i want them to be set automatically, as mentioned plugin do.

**Comment by webdiscus** [comment]
@ladown 
> If there are any possibilities how to configure preload for styles or scripts

This doesn't make sense because all scripts and styles must be manually specified in the Pug template anyway. Therefore, the additional preload attribute can be added directly by yourself.

> Тhe only thing that makes sense is the automatic addition of preloads for images, videos, fonts, etc, which was found  in the template or in CSS (fonts, images). I put this idea on the stack.

**Reply by ladown** [reply]
Okay, I get you. Thanks for the feedback
